crates/motoko/src/lib/vm.rs

```diff
@@ -988,374 +1018,409 @@ fn stack_cont(core: &mut Core, v: Value_) -> Result<Step, Interruption> {
+        match opaque_iter_next(core, &ptr)? {
+            None => {
+                core.cont = cont_value(Value::Unit);
                 Ok(Step {})
             }
-            // Updating for consistency with the `Call1` / `Call2` optimization
-            // Assign1(e2) => match &*v {
-            //     Value::Pointer(p) => exp_conts(core, Assign2(v), e2),
-            //     Value::Index(p, i) => exp_conts(core, Assign2(v), e2),
-            //     _ => Err(Interruption::TypeMismatch),
-            // },
-            Assign1(e2) => exp_conts(core, Assign2(v), &e2),
-            Assign2(v1) => match &*v1 {
-                Value::Pointer(p) => {
-                    store::mutate(core, p.clone(), v)?;
-                    core.cont = cont_value(Value::Unit);
-                    Ok(Step {})
-                }
-                Value::Index(p, i) => {
-                    store::mutate_index(core, p.clone(), i.fast_clone(), v)?;
-                    core.cont = cont_value(Value::Unit);
-                    Ok(Step {})
-                }
-                _ => Err(Interruption::TypeMismatch),
-            },
-            Idx1(e2) => exp_conts(core, Idx2(v), &e2),
-            Idx2(v1) => {
-                if let Some(Frame {
-                    cont: FrameCont::Assign1(_), // still need to evaluate RHS of assignment.
-                    ..
-                }) = core.stack.get(0)
-                {
-                    match &*v1 {
-                        Value::Pointer(p) => {
-                            // save array pointer and offset until after RHS is evaluated.
-                            core.cont = cont_value(Value::Index(p.clone(), v.fast_clone()));
-                            Ok(Step {})
-                        }
-                        Value::Dynamic(_) => Err(Interruption::Other(
-                            "Dynamic Rust value without a pointer".to_string(),
-                        )),
-                        _ => Err(Interruption::TypeMismatch),
-                    }
-                } else {
-                    let v1 = core.deref_value(v1)?;
-                    match (&*v1, &*v) {
-                        (Value::Array(_mut, a), Value::Nat(i)) => {
-                            let i = usize_from_biguint(i)?;
-                            core.cont = cont_value(
-                                (**a.get(i).ok_or(Interruption::IndexOutOfBounds)?).clone(),
-                            );
-                            Ok(Step {})
-                        }
-                        (Value::Dynamic(d), _) => {
-                            core.cont = cont_value((*d.dynamic().get_index(core, v)?).clone());
-                            Ok(Step {})
-                        }
-                        _ => Err(Interruption::TypeMismatch),
-                    }
-                }
-            }
-            Let(p, cont) => {
-                if let Some(env) = pattern_matches(core.env.clone(), p.as_ref().data_ref(), v) {
```

Comment:
This diff is confused, and thus, is confusing.  This old `Let` case has nothing to do with the case it "matches" below.

These are times that I wish each AST node in history had a unique ID, and these kinds of edits would be apparently small, instead of seeming large for trivial, formatting-related reasons.